.circleci/config.yml

```diff
@@ -253,6 +253,107 @@ jobs:
                 include_job_number_field: false
                 include_project_field: false
                 failure_message: ':red_circle: Scheduled tests failed on iOS device!'
+  android-build-react-native-bridge:
+    docker:
+      - image: circleci/node:10
+    steps:
+      - checkout
+      - checkout-submodules
+      - npm-install
+      - run:
+          name: Build JS Bundle
+          command: |
+            # Try to save memory when building the bundles by using only one
+            # worker.
+            #
+            # See:
+            #
+            # - https://stackoverflow.com/a/56027775/809944
+            # - https://facebook.github.io/metro/docs/configuration/#maxworkers
+            # - Build failure example: https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/10073/workflows/b8eb0104-c426-4208-b077-00caac394a62/jobs/53247
+            echo "module.exports = { maxWorkers: 1 };" > ./gutenberg/packages/react-native-bridge/metro.config.js
+            npm run bundle:android
+          # This command may take a long time, so prevent CircleCI from killing
+          # the build if it doesn't receive output for more than its default
+          # ten minutes.
+          no_output_timeout: 30m
+          environment:
+            # This build runs on a 4 GB RAM Docker Executor. To avoid running
+            # out of memory, set the max memory size of V8's old memory section
+            # to something less than that "to leave memory for other uses and
+            # avoid swapping".
+            #
+            # See:
+            #
+            # - https://nodejs.org/api/cli.html#cli_max_old_space_size_size_in_megabytes
+            # - https://support.circleci.com/hc/en-us/articles/360009208393-How-can-I-increase-the-max-memory-for-Node-
+            # - Build failure example: https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/10079/workflows/f11af227-783c-4304-a4bf-25f36a831ea9/jobs/53256
+            NODE_OPTIONS: --max_old_space_size=3072
```

Comment:
My other attempt to avoid running out of memory has been to pass this environment variable. I'm not even sure `npm` or Metro actually honor it or if maybe they pass other values to `node` when they call it under the hood.